entity-service-change-event-api/src/main/java/org/hypertrace/entity/service/change/event/api/EntityChangeEventGenerator.java

```diff
@@ -0,0 +1,17 @@
+package org.hypertrace.entity.service.change.event.api;
+
+import java.util.Map;
+import org.hypertrace.entity.data.service.v1.Entity;
+
+/** The interface Entity change event generator. */
+public interface EntityChangeEventGenerator {
+
+  /**
+   * Send change notification for any change to the entities
+   *
+   * @param existingEntityMap the map of existing entities
+   * @param updatedEntityMap the map of updated entities
+   */
+  void sendChangeNotification(
```

Comment:
this seems like a strange interface except for update. Can we have a cleaner create/delete method? Now that I think about it, even on updates why not just take a list - if the generator needs to put that into a map, that's an impl detail. From the interface, it's not clear whaat the key is (I'm guessing entity ID?)